### PR TITLE
Initial setup, the flow is there. 

### LINKED ISSUE #20: Set up simple SSG Markdown blog
Takes a folder of markdown blog posts and generates static site on compile.

Users only have to checkout a PR, upload a markdown file into the proper , and merge the 

Markdown can contain specific YAML:
Title, Description, Hero Image, etc

https://nextjs.org/docs/pages/building-your-application/configuring/mdx
https://vercel.com/templates/next.js/blog-starter-kit

### PR DESCRIPTION
**Pull request by sid597**
The website extract the blog from the folder, shows 3 of them, can see all the blogs, when clicked takes to the blog and renders it.

- [x] Check correctness of mdx doc parsing and rendering
- [x] UI seems a bit inconsistent from rest of the page for e.g the font size seems a bit off, not sure what text colouring to use
- [x] On new doc add rebuild and deploy

Addresses #20 

You can checkout the branch and build locally to test the functionality, ui & ux.

## Release Notes

- **New Features**
  - Dynamic loading of individual blog posts with structured error handling.
  - Blog index page displaying a list of all blog entries with metadata.
  - Latest blog updates section on the homepage showcasing recent posts.
  - Enhanced layout with a structured header and navigation system.
  - Template created for new blog posts with frontmatter instructions.

- **Bug Fixes**
  - Improved error handling for blog post retrieval and validation.

- **Documentation**
  - Introduced a new schema for blog posts to ensure data validation.
  - Added a Markdown template for creating new blog posts with guidelines.

- **Chores**
  - Updated dependencies to enhance functionality and maintainability.
  - Configured Tailwind CSS to include typography plugin for better styling options.

### CODE REVIEW

File: apps/website/app/blog/page.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+export default async function BlogIndex() {
+  const blogs = await getAllBlogs();
+
+  return (
+    <div className="min-h-screen bg-gray-50">
+      <div className="mx-auto max-w-6xl space-y-12 px-6 py-12">
+        <div className="rounded-xl bg-white p-8 shadow-md">
+          <div className="mb-8">
+            <h1 className="text-4xl font-bold text-gray-800">All Blog Posts</h1>
```
Comment (mdroidian) [inline apps/website/app/blog/page.tsx]: `text-primary` targets to the tailwind defined color, same as the main page

File: apps/website/app/blog/[slug]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+export default async function BlogPost({ params }) {
+  const { slug } = await params;
+  const { data, contentHtml } = await getBlog(slug);
+
+  return (
+    <div className="min-h-screen bg-gray-50 flex flex-col items-center px-6 py-12">
+      <div className="max-w-4xl w-full">
+        <header className="mb-8 text-center">
+          <h1 className="text-5xl font-bold leading-tight text-gray-800 mb-4">
```
Comment (mdroidian) [inline apps/website/app/blog/[slug]/page.tsx]: `text-primary` targets to the tailwind defined color, same as the main page

File: apps/website/package.json [hunk 3]
```diff
@@ -9,21 +9,27 @@
     "lint": "next lint"
   },
   "dependencies": {
+    "classnames": "^2.5.1",
+    "date-fns": "^3.6.0",
+    "gray-matter": "^4.0.3",
+    "next": "15.0.2",
+    "@repo/ui": "*",
+    "next": "^15.0.3",
```
Comment (mdroidian) [inline apps/website/package.json:15]: duplicate

**Comment by mdroidian** [comment]
Good start!  Here's some initial feedback.

It looks like the build script has some tight type validation which is failing currently.  If you do `turbo build --filter=discourse-graphs-website` you can repo the error locally.  Once the types are in there and it is building, Vercel will deploy it so that a version is live which we can share with the PI's: https://vercel.com/discourse-graphs/discourse-graph/58mY1PXGTkT9HptMiBGsPmu43KgM

For the type safety, I think we should use https://www.npmjs.com/package/zod.  I believe we could set this up in such a way that when anyone adds an `.md` file they will have type safe yaml entries.

A bit of a nit, but I might want to move the `/blogs` to co-locate them inside the `/blog` directory, maybe even rename to `/blog/posts`.

We could change the terminology from "blogs" to "updates" ("Latest Blog Posts" -> "Latest Updates"), which would give a more open ended range until different patterns of updates emerge. 

I would have expected to see a `package-lock.json` included in this since there were some packages added.  Did you notice one?

Also, there are some `prettier` formatting changes that get applied on save.  What IDE do you use?  Do you have prettier installed?

File: apps/website/app/page.tsx [hunk 4]
```diff
@@ -8,10 +8,32 @@ import {
...
+      };
+    })
+    .sort((a, b) => new Date(b.date) - new Date(a.date)) // Sort by date
+    .slice(0, 3); // Get the first 3 blogs
+}
+export default async function Home() {
+  const blogs = await getLatestBlogs();
   return (
     <div className={`min-h-screen bg-neutral-light ${inter.className}`}>
       <header className="flex flex-col items-center justify-between space-y-4 px-6 py-4 md:flex-row md:space-y-0">
```
Comment (mdroidian) [inline apps/website/app/page.tsx]: Let's move the header/nav to the root layout.tsx, that way the /blog and blog posts will inherit it.

We can also move the `bg-neutral-light` and any other classes that should wrap the body

**Comment by sid597** [comment]
Good catches Michael!

> It looks like the build script has some tight type validation which is failing currently. If you do `turbo build --filter=discourse-graphs-website` you can repo the error locally. Once the types are in there and it is building, Vercel will deploy it so that a version is live which we can share with the PI's: https://vercel.com/discourse-graphs/discourse-graph/58mY1PXGTkT9HptMiBGsPmu43KgM

Fixed, thanks for the command I was not aware of filter, it did not show any errors locally, check is failing from deployment pov I think. Will the deployment happen once we merge or can we deploy before merge like a test branch build preview kinda thing?

> For the type safety, I think we should use https://www.npmjs.com/package/zod. I believe we could set this up in such a way that when anyone adds an `.md` file they will have type safe yaml entries.

> A bit of a nit, but I might want to move the `/blogs` to co-locate them inside the `/blog` directory, maybe even rename to `/blog/posts`.

Both Done

> I would have expected to see a `package-lock.json` included in this since there were some packages added. Did you notice one?

Yeah it was there I forgot to commit because I did `git add .` while being in the `apps/website` path in terminal so missed the package-lock

> Also, there are some `prettier` formatting changes that get applied on save. What IDE do you use? Do you have prettier installed?

I use neovim which is currently not setup for typescript so did not have prettier or linting, will configure it now.

File: apps/website/app/blog/[slug]/page.tsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+export default async function BlogPost({ params }: Params) {
+  try {
+    const { slug } = await params;
+    const { data, contentHtml } = await getBlog(slug);
+
+    return (
+      <div className="flex flex-1 flex-col items-center bg-gray-50 px-6 py-12">
+        <div className="w-full max-w-4xl">
+          <header className="mb-8 text-center">
+            <h1 className="mb-4 text-5xl font-bold leading-tight text-gray-800 text-primary">
```
Comment (mdroidian) [inline apps/website/app/blog/[slug]/page.tsx]: image

File: apps/website/app/blog/page.tsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+export default async function BlogIndex() {
+  const blogs = await getAllBlogs();
+  return (
+    <div className="flex-1 bg-gray-50">
+      <div className="mx-auto max-w-6xl space-y-12 px-6 py-12">
+        <div className="rounded-xl bg-white p-8 shadow-md">
+          <div className="mb-8">
+            <h1 className="text-4xl font-bold text-gray-800 text-primary">
```
Comment (mdroidian) [inline apps/website/app/blog/page.tsx]: image

File: apps/website/app/blog/page.tsx [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+              All Blog Posts
```
Comment (mdroidian) [inline apps/website/app/blog/page.tsx]: Change to "All Updates"

File: apps/website/app/layout.tsx [hunk 8]
```diff
@@ -17,14 +20,58 @@ export const metadata: Metadata = {
...
+                Discourse Graphs
+              </span>
+            </Link>
+            <nav className="w-full md:w-auto">
+              <ul className="flex flex-wrap justify-center space-x-4 md:flex-nowrap md:space-x-8">
+                {[
+                  "About",
+                  "Resources",
+                  "Events",
+                  "Blog",
```
Comment (mdroidian) [inline apps/website/app/layout.tsx]: Change to "Updates": https://vercel.live/link/discoursegraphs.com?page=%2F%3FvercelThreadId%3DWyf9S&via=in-app-copy-link

File: apps/website/app/page.tsx [hunk 9]
```diff
@@ -461,6 +450,51 @@ export default function Home() {
             </CardContent>
           </Card>
 
+          {/* Blog Section */}
+          <Card id="blog" className="rounded-xl bg-white/50 p-8 shadow-md">
+            <CardHeader>
+              <CardTitle className="mb-8 text-4xl font-bold text-primary">
+                Latest Blog Posts
```
Comment (mdroidian) [inline apps/website/app/page.tsx]: Change to "Latest Updates"

File: apps/website/app/blog/page.tsx [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+            </h1>
+          </div>
+          <div>
+            <ul className="space-y-6">
+              {blogs.map((blog) => (
+                <li
+                  key={blog.slug}
+                  className="flex items-start justify-between border-b border-gray-200 pb-4"
```
Comment (mdroidian) [inline apps/website/app/blog/page.tsx]: we can remove the last bottom border with `last:border-b-0`

**Comment by mdroidian** [comment]
> Fixed, thanks for the command I was not aware of filter,

I actually just renamed the packages after that message 😅
So it would be `turbo build --filter=website` or `turbo dev --filter=roam`

> Will the deployment happen once we merge

I believe the deployment normally auto updates, but it looks like it was still waiting on an authorization.  We'll have to see how we can automate that.  I authorized it and it deployed to `Preview`. https://vercel.com/discourse-graphs/discourse-graph/khJ9tAJudwDFTobiJSCT6F9pEtFN

> I use neovim

Ah interesting.  We can look into this on Thursday.  If we can't get prettier configured we can add an `npm run format` script to do the formatting.

**Comment by sid597** [comment]
bug after mergin main to this branch, the error is same what you are mentioning in other pr. Looking into it if I can find anything

**Comment by sid597** [comment]
>Vercel for GitHub will deploy every push by default. This includes pushes and pull requests made to branches. This allows those working within the repository to preview changes made before they are pushed to production.



> The latest push to any pull request will automatically be made available at a unique [preview URL](https://vercel.com/docs/deployments/preview-deployments#preview-urls) based on the project name, branch, and team or username. These URLs will be provided through a comment on each pull request. Vercel also supports Comments on preview deployments made from PRs on GitHub. [Learn more about Comments on preview deployments in GitHub here](https://vercel.com/docs/deployments/preview-deployments#github-integration).

> If you receive a pull request from a fork of your repository that includes a change to the [vercel.json](https://vercel.com/docs/cli) file or the Project has [Environment Variables](https://vercel.com/docs/projects/environment-variables), Vercel will require authorization from you or a [Team Member](https://vercel.com/docs/teams-and-accounts/team-members-and-roles) to deploy the pull request.
This behavior protects you from leaking sensitive Project information.
You can disable [Git Fork Protection](https://vercel.com/docs/projects/overview#git-fork-protection) in the Security section of your Project Settings.

Pretty cool, I like vercel+turbo+nextjs combo makes things easy. Will we be moving roam package to vercel as well?

**Comment by sid597** [comment]
Also ready for final review

**Comment by sid597** [comment]
I think we should change "Ecosystem" to "Get Involved" or "Contact" because the navbar uses "Contact" then it takes to bottom of page and there is no "Contact" header which is confusing.

**Comment by mdroidian** [comment]
> Pretty cool, I like vercel+turbo+nextjs combo makes things easy. Will we be moving roam package to vercel as well?

The roam package is already on Vercel.  It uses the GitHub Actions to call [deploy](https://github.com/DiscourseGraphs/discourse-graph/blob/502d0f6c399fcadc01877af74f449110dbaa589d/apps/roam/scripts/deploy.ts#L4).  You can view the storage [here](https://vercel.com/discourse-graphs/~/stores/blob/store_6B4K1ntlti17rKF1/browser).

**Comment by mdroidian** [comment]
> I think we should change "Ecosystem" to "Get Involved" or "Contact" because the navbar uses "Contact" then it takes to bottom of page and there is no "Contact" header which is confusing.

I agree but lets bring this up with the team as this was copied from a couple versions of previous sites.

File: apps/website/app/page.tsx [hunk 11]
```diff
@@ -461,6 +450,51 @@ export default function Home() {
             </CardContent>
           </Card>
 
+          {/* Blog Section */}
```
Comment (mdroidian) [inline apps/website/app/page.tsx:425]: for Production we should probably add something like

```{blogs.length > 0 && (```

and delete the example posts

File: apps/website/app/blog/page.tsx [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+  return (
+    <div className="flex-1 bg-gray-50">
+      <div className="mx-auto max-w-6xl space-y-12 px-6 py-12">
+        <div className="rounded-xl bg-white p-8 shadow-md">
+          <div className="mb-8">
+            <h1 className="text-4xl font-bold text-primary">All Updates</h1>
+          </div>
+          <div>
+            <ul className="space-y-6">
+              {blogs.map((blog) => (
```
Comment (mdroidian) [inline apps/website/app/blog/page.tsx:15]: for Production we should probably add something like

```typescript
           {blogs.length === 0 ? (
              <p className="py-8 text-center text-gray-500">
                No updates yet! Check back soon. 😊
              </p>
            ) : (
```

and delete the example posts

File: apps/website/app/page.tsx [hunk 13]
```diff
@@ -461,6 +450,51 @@ export default function Home() {
...
+                Latest Updates
+              </CardTitle>
+            </CardHeader>
+            <CardContent>
+              <div className="space-y-6">
+                <ul className="space-y-6">
+                  {blogs.map((blog) => (
+                    <li
+                      key={blog.slug}
+                      className="flex items-start justify-between border-b border-gray-200 pb-4"
```
Comment (mdroidian) [inline apps/website/app/page.tsx]: add `last:border-b-0`

File: apps/website/app/page.tsx [hunk 14]
```diff
@@ -461,6 +450,51 @@ export default function Home() {
...
+                        by {blog.author}
+                      </div>
+                    </li>
+                  ))}
+                </ul>
+
+                <div className="mt-6 text-center">
+                  <Link
+                    href="/blog"
+                    className="inline-block rounded-md bg-primary px-4 py-2 text-lg font-semibold text-white transition hover:bg-primary/80"
```
Comment (mdroidian) [inline apps/website/app/page.tsx]: The text color on hover feels off.  We don't really have a color schema for this yet, so maybe just add `hover:text-white` for now.

**Comment by mdroidian** [comment]
@sid597 Looking good for the initial setup.  🚀

After these changes, we could merge, then wait for the first "blog post" to be added and walk the PI's through it.
Alternatively, we could also add a separate task for adding instructions.  Something like an `EXAMPLE.md`, publication status (?), and README.md instructions like

```markdown
## Creating Blog Posts

1. [Fork](https://docs.github.com/en/github/getting-started-with-github/fork-a-repo) and [clone](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/cloning-a-repository) the repository.
2. [Create a new branch](https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/managing-branches) for your update.
2. Copy `EXAMPLE.md` to create a new post
3. Rename the file to your desired URL slug (e.g., `my-new-post.md`)
4. Update the frontmatter and content
5. Set `published: true` when ready
6. Push your branch and open a pull request. 🚀
```

File: apps/website/app/blog/posts/EXAMPLE.md [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+date: "2024-01-01"
+author: "Devs"
+published: false # Set to true to make this post visible
+---
+
+# How to Create Blog Posts
+
+This is a template post that shows how to create new blog posts. Keep this file as a reference and create new posts by following these instructions.
+
+### Creating Blog Posts
```
Comment (mdroidian) [inline apps/website/app/blog/posts/EXAMPLE.md:12]: Good idea to put the instructions here 🔥

**Pull request merged**